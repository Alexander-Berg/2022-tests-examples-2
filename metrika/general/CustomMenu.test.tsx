import * as React from 'react';
import { shallow, mount } from 'enzyme';

import CustomMenu from './CustomMenu';

describe('CustomMenu', () => {
    it('renders', () => {
        const component = shallow(
            <CustomMenu>
                <CustomMenu.Title>Some Title</CustomMenu.Title>
                <CustomMenu.Divider />
                <CustomMenu.List>
                    <CustomMenu.Item>Some Item</CustomMenu.Item>
                    <CustomMenu.Item>
                        <CustomMenu.Section collapsed={true}>
                            Some section
                        </CustomMenu.Section>
                        <CustomMenu.Item level={1}>Inner item</CustomMenu.Item>
                        <CustomMenu.Item level={1} padding={20}>
                            Inner item with pad
                        </CustomMenu.Item>
                    </CustomMenu.Item>
                </CustomMenu.List>
            </CustomMenu>,
        );
        expect(component).toMatchSnapshot();
    });
});

describe('CustomMenu.Section', () => {
    it('renders and toggles', () => {
        const clickFn = jest.fn();
        const component = mount(
            <CustomMenu.Section onToggle={clickFn}>Section</CustomMenu.Section>,
        );

        component.find('div').forEach((wrapper) => wrapper.simulate('click'));

        expect(clickFn).toBeCalled();
    });
});
