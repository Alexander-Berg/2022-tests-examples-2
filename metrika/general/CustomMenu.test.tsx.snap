// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomMenu renders 1`] = `
<styled.div>
  <styled.div>
    Some Title
  </styled.div>
  <Styled(Component) />
  <Component>
    <ForwardRef>
      Some Item
    </ForwardRef>
    <ForwardRef>
      <Component
        collapsed={true}
      >
        Some section
      </Component>
      <ForwardRef
        level={1}
      >
        Inner item
      </ForwardRef>
      <ForwardRef
        level={1}
        padding={20}
      >
        Inner item with pad
      </ForwardRef>
    </ForwardRef>
  </Component>
</styled.div>
`;
